airflow/providers/dbt/cloud/utils/openlineage.py

```diff
@@ -0,0 +1,126 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import asyncio
+import re
+from contextlib import suppress
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from airflow.providers.dbt.cloud.hooks.dbt import DbtCloudHook
+    from airflow.providers.dbt.cloud.operators.dbt import DbtCloudRunJobOperator
+    from airflow.providers.dbt.cloud.sensors.dbt import DbtCloudJobRunSensor
+    from airflow.providers.openlineage.extractors.base import OperatorLineage
+
+
+def generate_openlineage_events_from_dbt_cloud_run(
```

Comment:
This method is pretty dense and would benefit from more documentation, especially around async parts

Reply:
I've simplified async code a bit and added more docs.